spec/unit/source/svn_spec.rb

```diff
@@ -61,6 +61,33 @@
       expect(environments[3].dirname).to eq 'robobutler'
     end
   end
+
+  describe "generating environments with branch filter exclude prefix" do
+    before do
+      allow(subject.svn_remote).to receive(:branches).and_return ['master', 'development', 'dev-test', 'dev', 'test_2']
+      subject.instance_variable_set(:@ignore_branch_prefixes, ['dev', 'test'])
+    end
+
+    let(:environments) { subject.generate_environments }
+
+    it "creates an environment for each branch not in ignore_branch_prefixes" do
+      expect(subject.generate_environments.size).to eq(2)
+    end
+
+    it "uses the branch name as the directory by default" do
+      expect(environments[0].name).to eq 'production'
+      expect(environments[1].name).to eq 'master'
+    end
+  end
+
+  describe "filtering branches with ignore prefixes" do
+    let(:branches) { ['master', 'development', 'production', 'dev-test', 'dev', 'test_2'] }
```

Comment:
Same comment as in the git about having a test case with dev/test in the middle or end of a valid branch name.